rules/framework/vfs_overlay.bzl

```diff
@@ -161,35 +180,76 @@ def _merge_vfs_infos(base, vfs_infos):
 
 # Internally the "vfs obj" is represented as a dictionary, which is keyed on
 # the name of the root. This is an opaque value to consumers
-def _make_vfs_info(roots):
+def _make_vfs_info(name, data):
     keys = {}
-    for root in roots:
-        name = root["name"]
-        keys[name] = root
+    keys[name] = data
```

Comment:
still makes sense to make it a sep. method now that it just do a keyvalue assignment?

Reply:
The reason we want a factory for the `vfs_info` is that it sets a clear boundary: callers if this method generally aren't concerned about _what_ the details of `vfs_info`. It also improves readability to localizing this logic by related other code. The simple function abstractions here also make it easier to refactor the code